doc/cug.tex

```diff
@@ -5852,8 +5918,35 @@ \subsection{Suite And Task Event Handling}
     \item `retry' - the task failed but will retry
     \item `execution timeout' - task execution timed out
 \end{myitemize}
-MESSAGE, if provided, describes what has happened, and TASKID identifies
-the task (\lstinline=NAME.CYCLE= for cycling tasks).
+
+The value of each setting should be a set of shell scripts or shell script
+templates (see below).
+
+Alternatively, task event handlers can be specified using the
+\lstinline=[[[events]]]handler= and the \lstinline=[[[events]]]handler events=
+settings, where the former is a set of shell scripts or shell script templates
+(see below) and the latter is a list of events when these commands should be
+invoked.
+
+Each task event handler can be specified as a list of shell scripts or a shell
+script templates.
+
+A shell script template may have any or all of these patterns which will be
+substituted with actual values:
+\begin{myitemize}
+    \item %(event)s: event name
+    \item %(suite)s: suite name
+    \item %(point)s: cycle point
+    \item %(name)s: task name
+    \item %(submit_num)s: submit number
+    \item %(id)s: task ID (i.e. %(name)s.%(point)s)
+    \item %(message)s: event message, if any
+\end{myitemize}
+
+Otherwise, the shell script will be appended with the following arguments:
```

Comment:
"Otherwise, the shell script will be called with the following command line arguments:"?